### PR TITLE
refactor(edgeless): split block-surface package

### PR DESCRIPTION
**Pull request by doodlewind**
Before:

```ts
import { SurfaceBlockModel } from '../../surface-block/index.js'
```

After:

```ts
import { SurfaceBlockModel } from '@blocksuite/affine-block-surface'
```

TODO: remove legacy `blocks/surface-block` after all of its references are migrated to `@blocksuite/affine-block-surface`.

### CODE REVIEW

File: packages/affine/block-surface/src/element-model/mindmap.ts [hunk 1]
```diff
@@ -0,0 +1,1087 @@
...
+
+export type SerializedMindmapElement = SerializedElement & {
+  children: Record<string, NodeDetail>;
+};
+
+type MindmapElementProps = BaseElementProps & {
+  children: Y.Map<NodeDetail>;
+};
+
+export class MindmapElementModel extends GfxGroupLikeElementModel<MindmapElementProps> {
```
Comment (Saul-Mirone) [inline packages/affine/block-surface/src/element-model/mindmap.ts:77]: I still believe this should be part of the `affine-model`

Comment (doodlewind) [reply packages/affine/block-surface/src/element-model/mindmap.ts:77]: Will address this in further PR.

File: packages/affine/block-surface/src/surface-model.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+        if (children?.size === 0) {
+          childrenMap.delete(id);
+        }
+      }
+    }
+  },
+} satisfies Record<string, MigrationRunner<typeof SurfaceBlockSchema>>;
+
+export const SurfaceBlockSchema = defineBlockSchema({
```
Comment (Saul-Mirone) [inline packages/affine/block-surface/src/surface-model.ts:132]: Same here.

Comment (doodlewind) [reply packages/affine/block-surface/src/surface-model.ts:132]: Would also leave this in new PR.

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**